### PR TITLE
Convert YAML files into Service Models/objects

### LINKED ISSUE #36: We need some ability to deserialize YAML into Models
We've got [some examples}(http://hackers.lookout.com/deploydb/internals.html#_services) where we need to define something that approximates a "model" in the DeployDB codebase.

Dropwizard itself has support for a single configuration file, but we need some ability to define YAML-backed models.

For example, `services/auditlog.yml` would eventually be translated into a `deploydb.models.Service` instance for auditlog.

I don't want this issue to include the handling of loading those yaml files necessarily, but the code to translate configuration-driven models from YAML into objects.

### PR DESCRIPTION
**Pull request by maheshkelkar**
Added Service object model. Added tests to read various YAML files
and generate Service objects.

Fixes #36

### CODE REVIEW

File: src/main/groovy/deploydb/models/Service.groovy [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package deploydb.models
+
+import com.google.common.collect.Lists
+import com.fasterxml.jackson.annotation.JsonProperty
+import javax.validation.constraints.NotNull
+import org.hibernate.validator.constraints.NotEmpty
+
+class Service {
```
Comment (rtyler) [inline src/main/groovy/deploydb/models/Service.groovy]: This class should have some javadocs associated with it and its members.

File: src/test/groovy/deploydb/models/ServiceSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+class ServiceWithArgsSpec extends Specification {
+
+    private final Validator validator = Validation.buildDefaultValidatorFactory().getValidator()
+    private final ConfigurationFactory<Service> factory =
+            new ConfigurationFactory<>(Service.class, validator, Jackson.newObjectMapper(), "dw")
+
+    def "Loading of valid service config file suceeds"() {
+        given:
+        ClassLoader classLoader = getClass().getClassLoader()
+        File validFile = new File(classLoader.getResource("services/test-valid.yml").toURI())
```
Comment (rtyler) [inline src/test/groovy/deploydb/models/ServiceSpec.groovy]: Why not refactor these two lines into a helper method in the class which returns a `File` for a given file? It's used in enough specifications

File: src/test/resources/services/test-invalid.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+name: Boop
```
Comment (rtyler) [inline src/test/resources/services/test-invalid.yml]: boop

File: src/main/groovy/deploydb/models/Service.groovy [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    @NotNull
+    @JsonProperty
+    String name
+
+    @NotEmpty
+    @JsonProperty
+    List<String> artifacts
```
Comment (rtyler) [inline src/main/groovy/deploydb/models/Service.groovy]: Can you add additional validation to ensure that the list of artifacts follows the `group:name` convention?

**Comment by rtyler** [comment]
Overall a good looking first pass!

File: src/main/groovy/deploydb/models/Service.groovy (hunk 1 above)
Comment (maheshkelkar) [inline src/main/groovy/deploydb/models/Service.groovy]: what do you mean by javadocs? comments or comments with tags?

File: src/main/groovy/deploydb/models/Service.groovy (hunk 4 above)
Comment (maheshkelkar) [inline src/main/groovy/deploydb/models/Service.groovy]: I have added max size of 10 for number of artifacts, pipelines and promotions

**Comment by maheshkelkar** [comment]
Took care of the code review comments #36

**Pull request merged**